src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -324,20 +324,37 @@ private function getMetadata($key)
     public function purge($url)
     {
         $key = $this->getCacheKey(Request::create($url));
+        $uri2 = 0 === strpos($uri, 'https://') 
+            ? preg_replace('https', 'http', $uri, 1) 
+            : preg_replace('http', 'https', $uri, 1);
+        $key2 = $this->getCacheKey(Request::create($url2));
 
         if (isset($this->locks[$key])) {
             flock($this->locks[$key], LOCK_UN);
             fclose($this->locks[$key]);
             unset($this->locks[$key]);
         }
 
+        $found = false;
         if (file_exists($path = $this->getPath($key))) {
             unlink($path);
 
-            return true;
+            $found = true;
         }
 
-        return false;
+         if (isset($this->locks[$key2])) {
```

Comment:
the actual deleting could be moved to a private function to avoid code repetition.